libraries/chain/include/eosio/chain/host_context.hpp

```diff
@@ -81,7 +87,19 @@ class host_context {
             return _iterator_to_object.size() - 1;
          }
 
+         // Invalid the iterator entry for the  given object pointer
+         void invalidate(const void* obj_ptr) {
+            auto itr = _object_to_iterator.find((const T*)obj_ptr);
+            if (itr == _object_to_iterator.end()) return;
+
+            EOS_ASSERT(0 <= itr->second && (size_t)itr->second < _iterator_to_object.size(), invalid_table_iterator, "iterator out of range");
+
+            _object_to_iterator.erase((const T*)obj_ptr);
+            _iterator_to_object[itr->second] = nullptr;
+         }
+
       private:
+         host_context&                                   _context;
          map<table_id_object::id_type, pair<const table_id_object*, int>> _table_cache;
```

Comment:
Do we need to do anything similar with the table cache? I would have guessed so but I haven't studied it yet

Reply:
I think it should be OK not to update the table cache. When all rows in a table are removed, even though the table cache still has an entry for the table, that entry will not be used anymore because no iterators exist for that table.